### PR TITLE
Release v1.2.0

### PR DESCRIPTION
**Pull request by gitbuda**
#### Major Updates

* Added Windows support.
* Added Apple support.
* Added compile-time warnings setup.
* Added `DiscardAll` and `FetchAll` in C++.

#### But Fixes and Other Changes

* Fixed type checking in C++ `ConstValue::ValueMap` method.
* Fixed pull error handling in the C++ `Client::Execute` method.
* Fixed various compile-time warnings.
* Fixed `mg_session_params_get_client_name` by renaming it to `mg_session_params_get_user_agent` (the one implemented in `src/mgclient.c`).
* Add advanced C++ example.

### CODE REVIEW

File: include/mgclient.h [hunk 1]
```diff
@@ -1233,7 +1233,7 @@ MGCLIENT_EXPORT const char *mg_session_params_get_username(
     const mg_session_params *);
 MGCLIENT_EXPORT const char *mg_session_params_get_password(
     const mg_session_params *);
-MGCLIENT_EXPORT const char *mg_session_params_get_client_name(
+MGCLIENT_EXPORT const char *mg_session_params_get_user_agent(
```
Comment (gitbuda) [inline include/mgclient.h:1236]: This never worked properly... The `..._client_name` doesn't exist in the implementation file.

File: examples/advanced.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        const auto props = node.properties();
+        std::string props_str =
+            std::accumulate(props.begin(), props.end(), std::string("{"),
+                            [](const std::string &acc, const auto &key_value) {
+                              const auto &key = key_value.first;
+                              const auto &value = key_value.second;
+                              std::string value_str;
+                              if (value.type() == mg::Value::Type::Int)
+                                value_str = std::to_string(value.ValueInt());
+                              if (value.type() == mg::Value::Type::String)
```
Comment (antonio2368) [inline examples/advanced.cpp:84]: else if
...
else if

Comment (antonio2368) [reply examples/advanced.cpp:84]: switch case is also okay with default having exit(1) and log

File: include/mgclient.hpp [hunk 3]
```diff
@@ -189,6 +195,19 @@ inline std::optional<std::vector<Value>> Client::FetchOne() {
   return values;
 }
 
+inline void Client::DiscardAll() {
+  while (FetchOne())
```
Comment (antonio2368) [inline include/mgclient.hpp:199]: Writing this smarter would also be a cool onboarding/nice to have feature :thinking: 
Like buffering data and calling PULL/DISCARD messages when necessary.

File: examples/advanced.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            labels.begin(), labels.end(), std::string(""),
+            [](const std::string &acc, const std::string_view value) {
+              return acc + ":" + std::string(value);
+            });
+        const auto props = node.properties();
+        std::string props_str =
+            std::accumulate(
+                props.begin(), props.end(), std::string("{"),
+                [](const std::string &acc, const auto &key_value) {
+                  const auto &key = key_value.first;
```
Comment (antonio2368) [inline examples/advanced.cpp:80]: `const auto &[key, value] = key_value;`

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**